### PR TITLE
Transporter

### PR DESCRIPTION
**Pull request by cathmer**
Add transporter as middle man.

### CODE REVIEW

File: packages/item-cc/src/item.controller.ts [hunk 1]
```diff
@@ -155,15 +164,67 @@ export class ItemController extends ConvectorController {
   ) {
     //first check if item exists
     let item = await Item.getOne(itemId);
+
+    if (item.proposalAccepted) {
+      throw new Error('This proposal was already accepted, this cannot be changed anymore.');
```
Comment (Tim-W) [inline packages/item-cc/src/item.controller.ts:169]: Does this mean an item can only be transferred once? Why is this needed? The solution before was to put the proposedOwner field to null after transfer

File: packages/item-cc/src/item.controller.ts [hunk 2]
```diff
@@ -155,15 +164,67 @@ export class ItemController extends ConvectorController {
...
     await ItemController.checkValidOwner(this.sender, item.proposedOwner);
 
     //If the proposal is accepted, change the item owner
     if (accept) {
-      item.itemHistory.push(new TransferEvent(item.proposedOwner, item.name, item.itemOwner));
-      item.itemOwner = item.proposedOwner;
+      item.proposalAccepted = true;
+    } else {
+      item.proposalAccepted = false;
+      item.proposedOwner = '';
```
Comment (Tim-W) [inline packages/item-cc/src/item.controller.ts:180]: This line should always be executed, whether it gets accepted or not

File: packages/item-cc/src/item.controller.ts [hunk 3]
```diff
@@ -196,6 +257,20 @@ export class ItemController extends ConvectorController {
     });
   }
 
+  @Invokable()
+  public async getTransportItems(
+    @Param(yup.string())
+      transporterId: string
+  ) {
+    await ItemController.checkValidOwner(this.sender, transporterId);
```
Comment (Tim-W) [inline packages/item-cc/src/item.controller.ts:265]: Why can only the transporter himself see the items?

File: packages/item-cc/src/item.controller.ts (hunk 1 above)
Comment (cathmer) [reply packages/item-cc/src/item.controller.ts:169]: No it means that if a proposal is accepted by the receiving party, he cannot unaccept it anymore (because the deal is then made, you cannot undo it anymore)

File: packages/item-cc/src/item.controller.ts (hunk 2 above)
Comment (cathmer) [reply packages/item-cc/src/item.controller.ts:180]: No because the item goes to the transporter first, so you don't change ownership yet, therefore you still need the proposedOwner otherwise the transporter does not know where to send the item

File: packages/item-cc/src/item.controller.ts (hunk 3 above)
Comment (cathmer) [reply packages/item-cc/src/item.controller.ts:265]: The idea is that with this function the transporter receives all items that he needs to transport, and only he can see exactly which items these are. Similarly to getParticipantProposals, where only the proposedOwner can see items that are proposed to him. 

The owner and the proposedOwner can still also see the item though.
